src/views/mainPage/components/buttonContainer.js

```diff
@@ -0,0 +1,31 @@
+import { restaurants } from "../../../constants/restaurantData.js";
+import restaurantList from "./restaurantList.js";
+import { $ } from "../../../utils/domHelpers.js";
+import button from "../../../components/@common/button.js";
+import buttonHandler from "../../../eventHandler/button.js";
+
+const buttonContainer = () => {
+  const $buttonContainer = $(".button-container");
+  $buttonContainer.innerHTML = `
+      ${button({
+        type: "button",
+        id: "cancel-button",
+        className: "secondary",
+        children: "취소하기",
+      })}
+      ${button({
+        type: "button",
+        id: "add-button",
+        className: "primary",
+        children: "추가하기",
+      })}
+    
+  `;
+
+  const $addButton = $("#add-button");
+  const $cancelButton = $("#cancel-button");
+
+  $addButton.addEventListener("click", buttonHandler);
+};
```

Comment:
다만 실제 서비스라면 유효하지 않은 셀렉터 하나로 인해 전체 어플리케이션이 마비되는 것을 방지하기 위해 $ 를 사용하는 렌더링 로직 모두에 아래와 같은 고차함수를 적용할 수 있을 것 같습니다.

```suggestion
const errorContainer = (component, fallbackComponent) => {
  try {
    component();
  } catch {
    fallbackComponent();
  }
}

const buttonContainer = errorContainer(() => {
  const $buttonContainer = $(".button-container");
  $buttonContainer.innerHTML = `
      ${button({
        type: "button",
        id: "cancel-button",
        className: "secondary",
        children: "취소하기",
      })}
      ${button({
        type: "button",
        id: "add-button",
        className: "primary",
        children: "추가하기",
      })}
    
  `);

  const $addButton = $("#add-button");
  const $cancelButton = $("#cancel-button");

  $addButton.addEventListener("click", buttonHandler);
}, () => {
  ...
});
```

Reply:
와 너무 좋은데요? 고차함수는 익숙하지 않아서 생각조차 못했었는데, 너무 좋은 방안인 것 같습니다 ㅎㅎ!

Reply:
4621ae2b43bfd5e185b29187e23ba57716f3fe82

Reply:
가볍게 제안드리는 정도였는데 실제로 적용까지 해주셨군요! 수고하셨습니다 익셉!